trin-core/src/portalnet/storage.rs

```diff
@@ -342,10 +344,19 @@ impl PortalStorage {
                     (":offset", offset.to_string().as_str()),
                     (":limit", limit.to_string().as_str()),
                 ],
-                |row| Ok(ContentKey { key: row.get(0)? }),
+                |row| {
+                    Ok({
+                        let row: String = row.get(0)?;
+                        let content_key: HistoryContentKey = serde_json::from_value(json!(row))
+                            .map_err(|err| {
+                                rusqlite::Error::InvalidParameterName(err.to_string())
```

Comment:
This is a hack to get around the fact that rusqlite doesn't support returning a custom error type. Added TODO and will create an issue to fix this.

Reply:
but we don't have to use a `rusqlite` error do we? can't we just map to an `anyhow` error here?

Reply:
In fact, we need it because this function is an argument to [query_map](https://docs.rs/rusqlite/latest/rusqlite/struct.Statement.html#method.query_map) and has to return `rusqlite::Result`.